platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/appcomm/AppCommunicationComponent.java

```diff
@@ -46,40 +39,22 @@ public class AppCommunicationComponent implements PlatformComponent {
     private static final Logger logger = LogManager.getLogger(AppCommunicationComponent.class);
 
     private final NotificationEngine notificationEngine;
-    /** A queue thread that asynchronously invokes NewLatestCompleteStateConsumers */
-    private final QueueThread<ReservedSignedState> asyncLatestCompleteStateQueue;
+    private final Predicate<ReservedSignedState> offerPredicate;
+
     /** The round of the latest state provided to the application */
     private long latestStateProvidedRound = ConsensusConstants.ROUND_UNDEFINED;
-    /**
-     * The size of the queue holding tasks for
-     * {@link com.swirlds.platform.components.state.output.NewLatestCompleteStateConsumer}s
-     */
-    private final AverageAndMax asyncLatestCompleteStateQueueSize;
 
     /**
      * Create a new instance
      *
-     * @param notificationEngine the notification engine
-     * @param context            the platform context
+     * @param notificationEngine    the notification engine
+     * @param offerPredicate        the offer to app communicator predicate
      */
     public AppCommunicationComponent(
-            @NonNull final NotificationEngine notificationEngine, @NonNull final PlatformContext context) {
+            @NonNull final NotificationEngine notificationEngine,
+            @NonNull final Predicate<ReservedSignedState> offerPredicate) {
```

Comment:
With the changes I mentioned, you shouldn't need to pass in this predicate anymore